src/background/tab-controller.ts

```diff
@@ -107,6 +107,12 @@ export class TabController {
         }
     };
 
+    private handleTabUpdateOnUrlHasChanged = (tabId: number, changeInfo: chrome.tabs.TabChangeInfo): void => {
```

Comment:
The naming here confused me a little on first reading; I think it might be more clear to make this new helper be named `handleTabUpdate` and rename the original thing to be named `onTabUrlUpdate` or something

Reply:
I changed it up a bit; let me know what you think!